lib/arclight/repository.rb

```diff
@@ -59,11 +59,18 @@ def self.all
       from_yaml(ENV['REPOSITORY_FILE'] || 'config/repositories.yml').values
     end
 
-    # Mimics ActiveRecord `find_by` behavior
+    # Mimics ActiveRecord dynamic `find_by` behavior for the slug or name
     #
-    # @param [String] `slug`
-    def self.find_by(slug)
-      all.select { |repo| repo.slug == slug }.first
+    # @param [String] `slug` or `name`
+    # @return [Repository]
+    def self.find_by(slug: nil, name: nil)
+      if slug
+        all.select { |repo| repo.slug == slug }.first
```

Comment:
My understanding is that `find` is to grab the record by its "ID" value. `find_by` and `where` were the two options I found for a way to search by name or by slug. Or are you suggesting that we implement a `find` that looks for matches in both the slug and name data?